### PR TITLE
Increase watchCache throughput

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/73958

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go [hunk 1]
```diff
@@ -359,6 +360,7 @@ func NewCacherFromConfig(config Config) *Cacher {
 				}
 			}, time.Second, stopCh,
 		)
+		watchCache.Stop()
```
Comment (jpbetz) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go]: put this as a defer with the others (above)?

**Review by jpbetz — COMMENTED** [review]
Intuitively this seems like a solid optimization. Would the be any value to writing a go benchmark or instrumenting the buffer and running a scale test or something tune the buffer size? (and measuring the performance boost?)

**Comment by hormes** [comment]
The eventHandler is:

```
func (c *Cacher) processEvent(event *watchCacheEvent) {
	// ...
	c.incoming <- *event
}
```

What is the benefit of adding a eventHandler buffer? It is equivalent to increasing the size of the incoming buffer.

**Comment by wojtek-t** [comment]
@hormes - yeah, actually that's true. This change was part of a larger patch that I was experimenting with where it had sense (but as a whole it didn't behave visibly better and was much more complicated).
So I'm closing this PR.

**Pull request closed without merging**